File: backend/internal/model/landing.go
// Package model contains request and response DTOs.
package model

// LandingStats holds the marketplace-wide metrics shown on the public landing page.
type LandingStats struct {
	AvailableProducts int64   `json:"available_products"`
	Users             int64   `json:"users"`
	AverageRating     float64 `json:"average_rating"`
	TotalReviews      int64   `json:"total_reviews"`
}

// LandingCategory is a category with its available-item count for the landing page.
type LandingCategory struct {
	Category string `json:"category"`
	Count    int64  `json:"count"`
}

// LandingResponse is the public payload powering the marketing landing page.
type LandingResponse struct {
	Stats      LandingStats         `json:"stats"`
	Categories []LandingCategory    `json:"categories"`
	Featured   []SearchItemResponse `json:"featured"`
}

File: frontend/src/types/landing.ts
import type { SearchItemResponse } from "@/types/item";

/**
 * Marketplace-wide metrics shown on the public landing page.
 * Matches model.LandingStats in the backend.
 */
interface LandingStats {
  available_products: number;
  users: number;
  average_rating: number;
  total_reviews: number;
}

/**
 * A category with its available-item count.
 * Matches model.LandingCategory in the backend.
 */
interface LandingCategory {
  category: string;
  count: number;
}

/**
 * Public payload returned by GET /api/landing.
 * Matches model.LandingResponse in the backend.
 */
interface LandingResponse {
  stats: LandingStats;
  categories: LandingCategory[];
  featured: SearchItemResponse[];
}

export type { LandingStats, LandingCategory, LandingResponse };
